TrainingXamarin/TrainingXamarin/TrainingXamarin/Introduce/IntrodoceViewModel.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Windows.Input;
+using TrainingXamarin.Model;
+using Xamarin.Forms;
+
+namespace TrainingXamarin.Introduce
+{
+    public class IntrodoceViewModel : INotifyPropertyChanged
+    {
+        private int _postition = 0;
+        private string _btnName = AppResources.action_next;
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public IntrodoceViewModel(ContentPage contentPage)
+        {
+
+            Items = new ObservableCollection<Intro>
+            {
+                new Intro(){ Image = "icon_calender.png",
+                    Content = AppResources.title_intro_first},
+                new Intro(){ Image = "icon_calender.png",
+                    Content = AppResources.title_intro_second}
+            };
+
+            OnNextClick = new Command((nothing) =>
+            {
+
+                if (_postition == Items.Count - 1)
+                {
+                    contentPage.Navigation.PushAsync(new MainPage());
+                }
+
+                if (_postition < Items.Count - 1)
+                {
+                    Position++;
+                }
+            });
+        }
+
+        public void PositionSelected(object sender, int position)
+        {
+            if (position == Items.Count - 1)
+            {
+                BtnName = AppResources.action_finish;
+            }
+            else
+            {
+                BtnName = AppResources.action_next;
+            }
+        }
+
+        public ObservableCollection<Intro> Items
+        {
+            get; set;
+        }
+
+        public int Position
+        {
+            set
+            {
+                if (_postition != value)
+                {
+                    _postition = value;
+
+                    if (PropertyChanged != null)
```

Comment:
e tạo 1 class BaseViewModel : INotifyPropertyChanged
trong đó có hàm
void pushPropertyChanged(string propertyName)
{
if (PropertyChanged != null)
                    {
                         PropertyChanged(this,
                         new PropertyChangedEventArgs(propertyName));
                    }
}